apps/web/vibes/soul/primitives/dynamic-form/schema.ts

```diff
@@ -0,0 +1,234 @@
+import { z } from 'zod';
+
+interface FormField {
+  name: string;
+  label?: string;
+  errors?: string[];
+  required?: boolean;
+}
+
+type RadioField = {
+  type: 'radio-group';
+  options: { label: string; value: string }[];
+  defaultValue?: string;
+} & FormField;
+
+type SelectField = {
+  type: 'select';
+  options: { label: string; value: string }[];
+  defaultValue?: string;
+} & FormField;
+
+type CheckboxField = {
+  type: 'checkbox';
+  defaultValue?: string;
+} & FormField;
+
+type CheckboxGroupField = {
+  type: 'checkbox-group';
+  options: { label: string; value: string }[];
+  defaultValue?: string[];
+} & FormField;
+
+type NumberInputField = {
+  type: 'number';
+  defaultValue?: string;
+  min?: number;
+  max?: number;
+  step?: number;
+} & FormField;
+
+type TextInputField = {
+  type: 'text';
+  defaultValue?: string;
+} & FormField;
+
+type EmailInputField = {
+  type: 'email';
+  defaultValue?: string;
+} & FormField;
+
+type TextAreaField = {
+  type: 'textarea';
+  defaultValue?: string;
+} & FormField;
+
+type DateField = {
+  type: 'date';
+  defaultValue?: string;
+  minDate?: string;
+  maxDate?: string;
+} & FormField;
+
+type SwatchRadioFieldOption =
+  | {
+      type: 'color';
+      value: string;
+      label: string;
+      color: string;
+      disabled?: boolean;
+    }
+  | {
+      type: 'image';
+      value: string;
+      label: string;
+      image: { src: string; alt: string };
+      disabled?: boolean;
+    };
+
+type SwatchRadioField = {
+  type: 'swatch-radio-group';
+  defaultValue?: string;
+  options: SwatchRadioFieldOption[];
+} & FormField;
+
+type CardRadioField = {
+  type: 'card-radio-group';
+  defaultValue?: string;
+  options: {
+    value: string;
+    label: string;
+    image: { src: string; alt: string };
+    disabled?: boolean;
+  }[];
+} & FormField;
+
+type ButtonRadioField = {
+  type: 'button-radio-group';
+  defaultValue?: string;
+  pattern?: string;
+  options: {
+    value: string;
+    label: string;
+    disabled?: boolean;
+  }[];
+} & FormField;
+
+type PasswordField = {
+  type: 'password';
+} & FormField;
+
+type ConfirmPasswordField = {
+  type: 'confirm-password';
+} & FormField;
```

Comment:
Is the reason for these two types of fields because the BigCommerce API has them and also to validate that one matches the other?